exporter/awskinesisexporter/internal/batch/encode_marshaler.go

```diff
@@ -0,0 +1,156 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package batch
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/model/otlp"
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awskinesisexporter/internal/key"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/zipkin/zipkinv2"
+)
+
+type batchMarshaller struct {
+	batchOptions []Option
+	partitioner  key.Partition
+
+	logsMarshaller    pdata.LogsMarshaler
+	tracesMarshaller  pdata.TracesMarshaler
+	metricsMarshaller pdata.MetricsMarshaler
+}
+
+var _ Encoder = (*batchMarshaller)(nil)
+
+func NewEncoder(named string, batchOptions ...Option) (Encoder, error) {
+	bm := &batchMarshaller{
+		batchOptions:      batchOptions,
+		partitioner:       key.Randomized,
+		logsMarshaller:    unsupported{},
+		tracesMarshaller:  unsupported{},
+		metricsMarshaller: unsupported{},
+	}
+	switch named {
+	case "zipkin-proto", "zipkin_proto", "zipkin/proto":
```

Comment:
Since these are new, documented values, do we need all these variations?

Reply:
I imagine not, I was considering having a map of encoding --> marshaller generator function but I wasn't sure on the nomenclature for defining the encoding type so I went for this approach.

Reply:
I would stick with just defining one for each of these (except for otlp / otlp-proto)